src/NHSD.BuyingCatalogue.Solutions.Persistence/Repositories/PriceRepository.cs

```diff
@@ -42,5 +64,11 @@ public async Task<IEnumerable<ICataloguePriceListResult>> GetPricesBySolutionIdQ
             return await _dbConnector.QueryAsync<CataloguePriceListResult>(ListCataloguePricesSql, cancellationToken,
                 new { solutionId });
         }
+
+        public async Task<IEnumerable<ICataloguePriceListResult>> GetPriceByPriceIdQueryAsync(int cataloguePriceId, CancellationToken cancellationToken)
```

Comment:
This will just return a single result right? So it should just return ICataloguePriceListResult, although if you're using that, it might be best to rename it

Reply:
Discussed  via call , IEnumerable is required.